bpfd/src/multiprog/tc.rs

```diff
@@ -144,15 +141,12 @@ impl TcDispatcher {
                 ..Default::default()
             },
         )?;
+
         let link = new_dispatcher.take_link(link_id)?;
         self.handle = Some(link.handle());
-        self.link = Some(link);
+        mem::forget(link);
```

Comment:
Just for my understanding... What's the difference between link and `link.handle` here? Would the link not get automatically dropped because we're keeping that reference to `link.handle`

Reply:
`link` is a `SchedClassifierLink` object, while `link.handle()` is just the value of the handle (a `u32`) of the tc dispatcher.  For example, in the following:
```
nfvsdn-04:~$ tc filter show dev mynet1 ingress
filter protocol all pref 50 bpf chain 0 
filter protocol all pref 50 bpf chain 0 handle 0x1 dispatcher direct-action not_in_hw id 681 tag 69ab5e4b570e07c0 
```
The handle is 0x1.  The way I'm attaching the dispatcher, the handle gets automatically assigned by the system, and I need to get it and save it so I can re-instantiate the link later when I need to detach it when it's being replaced by a new dispatcher.  Getting the handle doesn't cause the link to get dropped or prevent the link from being dropped.  When I used to save the link with `self.link = Some(link);`, that prevented the link from getting dropped when link went out of scope.  It also prevented it from getting dropped when bpfd exited because the destructor for TC links (and others) wasn't working right.  However, that got fixed, so then the link started getting dropped when bpfd exited.

Reply:
> When you do this.... Has the link stayed up and running on the system in the meantime because it's pinned and you just need to "re-instantiate a reference to it in memory in order to actually do things with it?

Yes.